### ISSUE TITLE
Voronoi diagram with missing cells

### ISSUE THREAD

**Issue by michaelmaalouly** [glotzerlab/freud#1295]
### Description

Hello, 

I am trying to create a Voronoi diagram from a lagrangian simulation where I have particles. The expectations for the particles in terms of the PDF of the Voronoi cell areas are good; however when I look at the diagram I see some empty spaces. 

Here is an example:
![Figure_1_test](https://github.com/user-attachments/assets/56959ffe-7e47-466c-bb6d-12497bd7c670)

### Steps to Reproduce

```python
L=2*pi
path_npz=appendpath+'lagr/output/npz/'
coord_tra=np.load(path_npz+'./xycoord.%.2d.npz'%(irun))
xtra=coord_tra['x']
ytra=coord_tra['y']
flatxy=zip_xy(xtra,ytra,ifr)
voro = freud.locality.Voronoi()
flatxy3d=flat2dto3d_shift(flatxy,-pi)

box = freud.box.Box.square(L)

voro.compute((box, flatxy3d))
vol=voro.volumes
# Plot Voronoi with points and a custom cmap
plt.figure()
ax = plt.gca()
voro.plot(ax=ax, cmap="RdBu")
ax.scatter(flatxy3d[:, 0], flatxy3d[:, 1], s=2, c="k")
plt.savefig('./lagr/output/img/vordiag.png')
plt.show()
```

### Error Output

```shell
There is no output error, just a visualization issue.
```

### freud Version

v3.1.0

### Python Version

v3.13.0

### System Platform

MacOS

### Installation method

Download from conda-forge

### Developer

None

**Comment by DomFijan** [comment]
I see that you are converting your points from 2D to 3D. Can you confirm that this is done correctly in your snippet? z values should all be zero if system is 2D.

**Comment by michaelmaalouly** [comment]
Yes that is exactly what is done with all the values of z set to zero.

**Comment by DomFijan** [comment]
Could you provide a minimal example of the bug? I cannot reproduce it locally.
